### PR TITLE
Blake3 PoW

### PR DESCRIPTION
**Pull request by wizeguyy**
This updates the manager to use the new Blake3 consensus engine for mining. Must merge after the associated PR:
https://github.com/spruce-solutions/go-quai/pull/252

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/spruce-solutions/quai-manager
 
 go 1.16
 
+replace github.com/spruce-solutions/go-quai => /Users/wizeguyy/repos/go-quai
```
Comment (shreekarashastry) [inline go.mod:5]: remove this before merging

Comment (wizeguyy) [reply go.mod:5]: Good catch!

Comment (wizeguyy) [reply go.mod]: Fixed with force-push

File: manager/main.go [hunk 2]
```diff
@@ -168,16 +168,17 @@ func main() {
...
-		PowMode:       ethash.ModeNormal,
-		CachesInMem:   3,
-		DatasetsInMem: 1,
+	blake3Config := blake3.Config{
+		MiningThreads: 0,
+		NotifyFull:    true,
+	}
+	log.Println("Building RandomX dataset")
+	blake3Engine, err := blake3.New(blake3Config, nil, false)
+	if nil != err {
```
Comment (BryceAgent) [inline manager/main.go:177]: I think "if err != nil" is the more typical convention, altho we're not perfectly consistent on it.

**Review by BryceAgent — COMMENTED** [review]
No issues.

File: manager/main.go (hunk 2 above)
Comment (wizeguyy) [reply manager/main.go:177]: This style is actually intentional, as a common best practice for code safety (see [Yoda Conditions](https://en.wikipedia.org/wiki/Yoda_conditions)). Although Golang's assignment rules prevent _most_ of that class of errors anyways, so I'll concede Yoda notation isn't as important in Go projects.

That said... if we wanted to specify a different style, I can get on board 👍

Comment (alanorwick) [reply manager/main.go:177]: @wizeguyy @BryceAgent Please use if `err != nil` we follow go fmt as described here: https://go.dev/doc/effective_go#formatting

Comment (wizeguyy) [reply manager/main.go]: Fixed 👍

**Pull request merged**